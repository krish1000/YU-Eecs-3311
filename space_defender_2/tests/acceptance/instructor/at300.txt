--Test enemy fighter fire prempt.
toggle_debug_mode

play(5,10,1,38,38,38,38)
setup_select(1)
setup_next(1)
setup_select(4)
setup_next(1)
setup_select(3)
setup_next(1)
setup_select(5)
setup_next(1)
setup_next(1)

fire
move(A,2)
fire
fire
move(A,4)
fire
fire
