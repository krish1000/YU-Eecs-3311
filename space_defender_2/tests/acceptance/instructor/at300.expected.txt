  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,10,1,38,38,38,38)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(1)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_next(1)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:None
->setup_select(4)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:Heavy
->setup_next(1)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Standard
->setup_select(3)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Armoured
->setup_next(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(5)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
->setup_next(1)
  state:setup summary, debug, ok
  Weapon Selected:Standard
  Armour Selected:Heavy
  Engine Selected:Armoured
  Power Selected:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:260/260, energy:110/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[C,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:260/260, energy:105/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[C,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,10]
  Projectile:
    [-1,*]->damage:70, move:5, location:[C,2]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [C,1].
      A friendly projectile(id:-1) spawns at location [C,2].
  Enemy Action:
  Natural Enemy Spawn:
    A Fighter(id:1) spawns at location [A,10].
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  _  _  _  _  _  _  F
    B _  _  _  _  _  _  _  _  _  _
    C S  *  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,2)
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:260/260, energy:76/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,9]
    [2,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:T, location:[A,10]
  Projectile:
    [-1,*]->damage:70, move:5, location:[C,7]
    [-2,<]->damage:50, move:6, location:[A,8]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [C,2] -> [C,7]
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [A,2]
  Enemy Action:
    A Fighter(id:1) moves: [A,10] -> [A,9]
      A enemy projectile(id:-2) spawns at location [A,8].
  Natural Enemy Spawn:
    A Fighter(id:2) spawns at location [A,10].
      1  2  3  4  5  6  7  8  9 10
    A _  S  _  _  _  _  _  <  F  F
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  *  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:227/260, energy:75/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,8]
    [2,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,9]
  Projectile:
    [-3,*]->damage:70, move:5, location:[A,3]
    [-4,<]->damage:50, move:6, location:[A,7]
  Friendly Projectile Action:
    A friendly projectile(id:-1) moves: [C,7] -> out of board
  Enemy Projectile Action:
    A enemy projectile(id:-2) moves: [A,8] -> [A,2]
      The projectile collides with Starfighter(id:0) at location [A,2], dealing 37 damage.
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-3) spawns at location [A,3].
  Enemy Action:
    A Fighter(id:1) gains 1 armour.
    A Fighter(id:2) gains 1 armour.
    A Fighter(id:1) moves: [A,9] -> [A,8]
      A enemy projectile(id:-4) spawns at location [A,7].
    A Fighter(id:2) moves: [A,10] -> [A,9]
      A enemy projectile(id:-5) spawns at location [A,8].
      The projectile collides with Fighter(id:1) at location [A,8], healing 50 damage.
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  _  <  F  F  _
    B _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:231/260, energy:74/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,2]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:12, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,7]
    [2,F]->health:150/150, Regen:5, Armour:12, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,8]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:T, location:[B,10]
  Projectile:
    [-6,*]->damage:70, move:5, location:[A,3]
    [-7,<]->damage:50, move:6, location:[A,6]
  Friendly Projectile Action:
    A friendly projectile(id:-3) moves: [A,3] -> [A,8]
      The projectile collides with enemy projectile(id:-4) at location [A,7], negating damage.
      The projectile collides with Fighter(id:1) at location [A,8], dealing 9 damage.
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,2].
      A friendly projectile(id:-6) spawns at location [A,3].
  Enemy Action:
    A Fighter(id:1) gains 1 armour.
    A Fighter(id:2) gains 1 armour.
    A Fighter(id:1) moves: [A,8] -> [A,7]
      A enemy projectile(id:-7) spawns at location [A,6].
    A Fighter(id:2) moves: [A,9] -> [A,8]
      A enemy projectile(id:-8) spawns at location [A,7].
      The projectile collides with Fighter(id:1) at location [A,7], healing 50 damage.
  Natural Enemy Spawn:
    A Fighter(id:3) spawns at location [B,10].
      1  2  3  4  5  6  7  8  9 10
    A _  S  *  _  _  <  F  F  _  _
    B _  _  _  _  _  _  _  _  _  F
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->move(A,4)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:235/260, energy:56/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,4]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:12, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,6]
    [2,F]->health:150/150, Regen:5, Armour:12, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,7]
    [3,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[B,9]
  Projectile:
    [-9,<]->damage:50, move:6, location:[A,5]
    [-11,<]->damage:50, move:6, location:[B,8]
  Friendly Projectile Action:
    A friendly projectile(id:-6) moves: [A,3] -> [A,7]
      The projectile collides with enemy projectile(id:-7) at location [A,6], negating damage.
      The projectile collides with Fighter(id:1) at location [A,7], dealing 8 damage.
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [A,2] -> [A,4]
  Enemy Action:
    A Fighter(id:1) moves: [A,7] -> [A,6]
      A enemy projectile(id:-9) spawns at location [A,5].
    A Fighter(id:2) moves: [A,8] -> [A,7]
      A enemy projectile(id:-10) spawns at location [A,6].
      The projectile collides with Fighter(id:1) at location [A,6], healing 50 damage.
    A Fighter(id:3) moves: [B,10] -> [B,9]
      A enemy projectile(id:-11) spawns at location [B,8].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  S  <  F  F  _  _  _
    B _  _  _  _  _  _  _  <  F  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(6.0), debug, ok
  Starfighter:
    [0,S]->health:165/260, energy:55/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,4]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:0
  Enemy:
    [1,F]->health:143/150, Regen:5, Armour:13, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,5]
    [2,F]->health:150/150, Regen:5, Armour:13, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,6]
    [3,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[B,8]
  Projectile:
    [-11,<]->damage:50, move:6, location:[B,2]
    [-15,<]->damage:50, move:6, location:[B,7]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-9) moves: [A,5] -> [A,4]
      The projectile collides with Starfighter(id:0) at location [A,4], dealing 37 damage.
    A enemy projectile(id:-11) moves: [B,8] -> [B,2]
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,4].
      A friendly projectile(id:-12) spawns at location [A,5].
  Enemy Action:
    A Fighter(id:1) gains 1 armour.
    A Fighter(id:2) gains 1 armour.
    A Fighter(id:3) gains 1 armour.
    A Fighter(id:1) moves: [A,6] -> [A,5]
      The Fighter collides with friendly projectile(id:-12) at location [A,5], taking 57 damage.
      A enemy projectile(id:-13) spawns at location [A,4].
      The projectile collides with Starfighter(id:0) at location [A,4], dealing 37 damage.
    A Fighter(id:2) moves: [A,7] -> [A,6]
      A enemy projectile(id:-14) spawns at location [A,5].
      The projectile collides with Fighter(id:1) at location [A,5], healing 50 damage.
    A Fighter(id:3) moves: [B,9] -> [B,8]
      A enemy projectile(id:-15) spawns at location [B,7].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  S  F  F  _  _  _  _
    B _  <  _  _  _  _  <  F  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(7.0), debug, ok
  Starfighter:
    [0,S]->health:41/260, energy:54/110, Regen:4/4, Armour:13, Vision:7, Move:4, Move Cost:11, location:[A,4]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
      score:3
  Enemy:
    [2,F]->health:150/150, Regen:5, Armour:14, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,5]
    [3,F]->health:150/150, Regen:5, Armour:12, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[B,7]
  Projectile:
    [-15,<]->damage:50, move:6, location:[B,1]
    [-18,<]->damage:50, move:6, location:[B,6]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-11) moves: [B,2] -> out of board
    A enemy projectile(id:-15) moves: [B,7] -> [B,1]
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,4].
      A friendly projectile(id:-16) spawns at location [A,5].
      The projectile collides with Fighter(id:1) at location [A,5], dealing 57 damage.
  Enemy Action:
    A Fighter(id:1) gains 1 armour.
    A Fighter(id:2) gains 1 armour.
    A Fighter(id:3) gains 1 armour.
    A Fighter(id:1) moves: [A,5] -> [A,4]
      The Fighter collides with Starfighter(id:0) at location [A,4], trading 91 damage.
      The Fighter at location [A,4] has been destroyed.
    A Fighter(id:2) moves: [A,6] -> [A,5]
      A enemy projectile(id:-17) spawns at location [A,4].
      The projectile collides with Starfighter(id:0) at location [A,4], dealing 37 damage.
    A Fighter(id:3) moves: [B,8] -> [B,7]
      A enemy projectile(id:-18) spawns at location [B,6].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10
    A _  _  _  S  F  _  _  _  _  _
    B <  _  _  _  _  <  F  _  _  _
    C _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _
